### PR TITLE
Add missing properties for Vmss zone GA

### PR DESCRIPTION
**Pull request by wangjun198389**
Updated compute spec to include missing properties zoneBalance and platformFaultDomainCount in VirtualMachineScaleSetProperties. We are using 2017-12-01 api version which is zone GA version. Both properties are immutable and optional. These properties are only for zone scenario, which is in public preview now.

### CODE REVIEW

File: specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json [hunk 1]
```diff
@@ -6314,6 +6314,15 @@
         "singlePlacementGroup": {
           "type": "boolean",
           "description": "When true this limits the scale set to a single placement group, of max size 100 virtual machines."
+        },
+        "zoneBalance": {
+          "type": "boolean",
+          "description": "Whether to force stictly even Virtual Machine distribution cross x-zones in case there is zone outage."
```
Comment (anuchandy) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json]: is x-zones some domain specific name?

**Comment by anuchandy** [comment]
@wangjun198389 you mentioned these are immutable properties, do you mean they are `readonly` if so please mark so. Or you mean they can set only once and cannot be changed later?

File: specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json (hunk 1 above)
Comment (wangjun198389) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json]: Somehow. "x-zones" or "zones redundant" or "cross zones"  VMScaleSet.

**Comment by wangjun198389** [comment]
@anuchandy Those new properties can set only once when creating, cannot be changed later.

**Comment by anuchandy** [comment]
@wangjun198389 sounds good, just one more question from my side - let's say we generate SDK from this PR `ClientNew` and used it to set these two properties as part of creation of `resource1`, suppose somebody is using current version of SDK say `ClientOld` (generated from the same api version spec in master) to do a PUT (update) on `resource1`, since these two properties does not exists in the model, payload will not contain them. In this case how server behaves? does it throw error?

**Comment by wangjun198389** [comment]
@anuchandy old client can put(update) successfully. Server side treat it as no change for these two new properties.

**Comment by anuchandy** [comment]
@wangjun198389 thanks. @ravbhatnagar looks good from SDK side, any comments from ARM side?

**Comment by ravbhatnagar** [comment]
new properties typically require a new api-version. but based on above comments, these properties are immutable. So its fine. Signing off!

**Review by anuchandy — APPROVED** [review]

**Pull request merged**